src/core/geometry/qgssfcgalgeometry.h

```diff
@@ -0,0 +1,516 @@
+
+    /**
+     * Cover test on 2D or 3D geometries
+     * Checks if this covers \a otherGeom.
+     * A 3D covers test is conducted when at least one geometry is 3D; otherwise, a 2D covers test is carried out.
+     *
+     * \param otherGeom second geometry to perform the operation
+     * \param errorMsg Error message returned by SFGCAL
+     */
+    bool covers( const QgsSfcgalGeometry &otherGeom, QString *errorMsg = nullptr ) const;
+
+    /**
+     * Calculate a 3D buffer where all points are at \a distance from the original geometry.
+     * A negative distance shrinks the geometry rather than expanding it.
+     * It is limited to Point and LineString.
+     * If the operation fails, a null pointer is returned.
+     *
+     * \param radius the buffer radius
+     * \param segments the number of segments to use for approximating curved
+     * \param joinStyle3D the type of buffer to compute
+     * \param errorMsg Error message returned by SFGCAL
+     */
+    QgsSfcgalGeometry *buffer3D( double radius, int segments, Qgis::JoinStyle3D joinStyle3D, QString *errorMsg = nullptr ) const;
+
+    /**
+     * Calculate a 2D buffer where all points are at \a distance from the original geometry.
+     * A negative distance shrinks the geometry rather than expanding it.
+     * If the operation fails, a null pointer is returned.
+     *
+     * \param radius the buffer radius
+     * \param segments the number of segments to use for approximating curved
+     * \param joinStyle the type of buffer to compute. Only round is supported.
+     * \param errorMsg Error message returned by SFGCAL
+     */
+    QgsSfcgalGeometry *buffer2D( double radius, int segments, Qgis::JoinStyle joinStyle, QString *errorMsg ) const;
+
+    /**
+     * Simplifies a geometry using the CGAL algorithm
+     *
+     * \param tolerance The distance (in geometry unit) threshold
+     * \param preserveTopology Whether to preserve topology during simplification
+     * \param errorMsg Error message returned by SFGCAL
+     */
+    QgsSfcgalGeometry *simplify( double tolerance, bool preserveTopology, QString *errorMsg = nullptr ) const;
+
+    /**
+     * Calculate an extrusion of the original geometry.
+     * If the operation fails, a null pointer is returned.
+     *
+     * \param extrusion extrusion vector (2D or 3D)
+     * \param errorMsg Error message returned by SFGCAL
+     */
+    QgsSfcgalGeometry *extrude( const QgsPoint &extrusion, QString *errorMsg = nullptr ) const;
+
+
+#ifndef SIP_RUN
+    /**
+     * Cast the \a geom to the exact underlying QGIS geometry.
+     * Should be used, for example, by qgsgeometry_cast<QgsPoint *>( geometry ) or by qgsgeometry_cast<QgsPolygon *>( geometry ).
+     *
+     * \note Not available in Python. Objects will be automatically be converted to the appropriate target type.
+     */
+    inline static const QgsAbstractGeometry *cast( const QgsSfcgalGeometry *geom )
```

Comment:
Please drop this method -- it is not memory safe. Here you are returning a new object, yet everywhere else qgsgeometry_cast does not.